src/languages/php.js

```diff
@@ -18,6 +18,28 @@ export default function(hljs) {
       // Perl at all like $ident$, @ident@, etc.
       `(?![A-Za-z0-9])(?![$])`
   };
+  const CONSTANT = {
+    className: "variable",
+    relevance: 0,
+    variants: [
+      {
+        begin: /const\s*/,
+        end: /\w+/,
+        excludeBegin: true
```

Comment:
Nope, far too broad as this would match `const 100 lines of source code here \w+` not just `const \w+`.  Please see the docs and use a multi-matcher for this: https://highlightjs.readthedocs.io/en/latest/mode-reference.html#beginscope

Reply:
See `wren` and other newer grammars for examples.

Reply:
\w+ will stop matching on first `;` or `=` which are mandatory in PHP or any non-word character, like space. I also added `relevance: 0` to not [detect](https://github.com/highlightjs/highlight.js/blob/main/test/detect/javascript/short-plain.txt)  as PHP.

I will look at beginscope tomorrow.

Reply:
> which are mandatory in PHP

Except this might not be parsing PHP at all... autodetect works by just parsing with everything and seeing what "works"... (and there is also no guarantee we're given VALID PHP).  Hence every rule must be as "tight" as possible.

Reply:
> relevance: 0

This is probably too low since `const` itself is a REAL keyword worth one point, so I'd remove it and let the default relevance of 1 apply (to the keyword, not the variable name).

Reply:
I thought `relevance: 0` only applies to const name and `const` as a keyword still has relevance 1.

Reply:
Your matching it so the keyword engine will never see it. Unless excludeBegin does the trick.  I don't actually know off the top of my head. The style you're using is a bit different. 

Once you use multi match though you'll definitely need to handle the keyword yourself so no relevance hacks should be needed.

Reply:
I removed `relevance` and changed all `className` to `scope`.